crates/turbopack-mdx/src/lib.rs

```diff
@@ -0,0 +1,192 @@
+#![feature(min_specialization)]
+
+use anyhow::{anyhow, Result};
+use mdxjs::compile;
+use turbo_tasks::{primitives::StringVc, Value, ValueToString, ValueToStringVc};
+use turbo_tasks_fs::{rope::Rope, File, FileContent, FileSystemPathVc};
+use turbopack_core::{
+    asset::{Asset, AssetContent, AssetContentVc, AssetVc},
+    chunk::{ChunkItem, ChunkItemVc, ChunkVc, ChunkableAsset, ChunkableAssetVc, ChunkingContextVc},
+    context::AssetContextVc,
+    reference::AssetReferencesVc,
+    resolve::origin::{ResolveOrigin, ResolveOriginVc},
+    virtual_asset::VirtualAssetVc,
+};
+use turbopack_ecmascript::{
+    chunk::{
+        EcmascriptChunkItem, EcmascriptChunkItemContentVc, EcmascriptChunkItemVc,
+        EcmascriptChunkPlaceable, EcmascriptChunkPlaceableVc, EcmascriptChunkVc,
+    },
+    AnalyzeEcmascriptModuleResultVc, EcmascriptInputTransformsVc, EcmascriptModuleAssetType,
+    EcmascriptModuleAssetVc,
+};
+
+#[turbo_tasks::value]
+#[derive(Clone, Copy)]
+pub struct MdxModuleAsset {
+    pub source: AssetVc,
+    pub context: AssetContextVc,
+    pub transforms: EcmascriptInputTransformsVc,
+}
+
+/// MDX components should be treated as normal j|tsx components to analyze
+/// its imports or run ecma transforms,
+/// only difference is it is not a valid ecmascript AST we
+/// can't pass it forward directly. Internally creates an jsx from mdx
+/// via mdxrs, then pass it through existing ecmascript analyzer.
+async fn interop_ecma_asset_from_mdx(
+    current_context: &MdxModuleAssetVc,
+) -> Result<EcmascriptModuleAssetVc> {
+    let content = current_context.content();
+    let this = current_context.await?;
+
+    if let AssetContent::File(file) = &*content.await? {
```

Comment:
```suggestion
    let AssetContent::File(file) = &*content.await? else {
        bail!("Unexpected mdx asset content");
    };
```

We can now use the following to avoid right-ward indentation shifts.